### PR TITLE
Feature/#612 get user list

### LINKED ISSUE #612: Get user list from data to domain

### PR DESCRIPTION
**Pull request by abdulazizacc**
**In this PR we implemented the get user functionality from dataSource to repo to domain**

### CODE REVIEW

File: domain/src/main/java/com/baghdad/domain/usecase/savedList/GetSavedListUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.baghdad.domain.usecase.savedList
+
+import com.baghdad.domain.model.PagedResult
+import com.baghdad.domain.repository.SavedListRepository
+
+class GetSavedListUseCase(
```
Comment (MuhammedEdrees) [inline domain/src/main/java/com/baghdad/domain/usecase/savedList/GetSavedListUseCase.kt]: rename to `GetSavedListsUseCase`

File: domain/src/main/java/com/baghdad/domain/usecase/savedList/GetSavedListUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    private val savedListRepository: SavedListRepository
+) {
+    suspend operator fun invoke(
+        page: Int,
+        sessionId: String
```
Comment (MuhammedEdrees) [inline domain/src/main/java/com/baghdad/domain/usecase/savedList/GetSavedListUseCase.kt]: add pageSize and remove sessionId, it will be retrieved directly in repository

File: remote_datasource/src/main/java/com/baghdad/remoteDataSource/apiService/SavedListApiService.kt [hunk 3]
```diff
@@ -1,3 +1,20 @@
...
+    @Authenticated
+    @GET(LISTS_ENDPOINT)
+    suspend fun getSavedLists(
+        @Query("page") page: Int,
+        @Query("session_id") sessionId: String
+    ): Response<UserListsResponse>
+
+    companion object {
+        private const val LISTS_ENDPOINT = "account/{account_id}/lists"
+    }
```
Comment (MuhammedEdrees) [inline remote_datasource/src/main/java/com/baghdad/remoteDataSource/apiService/SavedListApiService.kt]: accountId is added as path parameter here, We need to test if we can use only sessionId in which case we will set it to null or if we must use accoutnId then it will need to be passed from remote source

File: remote_datasource/src/main/java/com/baghdad/remoteDataSource/mapper/SavedListsResponseMapper.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import com.baghdad.remoteDataSource.util.getNextKey
+import com.baghdad.remoteDataSource.util.getPreviousKey
+import com.baghdad.repository.model.PagedResultDto
+
+import com.baghdad.repository.model.SavedListDto
+
+
+fun UserListsResponse.toPagedSavedListsDtos(): PagedResultDto<SavedListDto> {
+    return PagedResultDto(
+        data = this.results?.map { it.toSavedListDto() } ?: emptyList(),
```
Comment (MuhammedEdrees) [inline remote_datasource/src/main/java/com/baghdad/remoteDataSource/mapper/SavedListsResponseMapper.kt]: Just to be safe, add `takeIf` here similar to other mappers to filter null ids.

**Review by MuhammedEdrees — CHANGES_REQUESTED** [review]
Well done so far

**Review by MuhammedEdrees — DISMISSED** [review]

**Review by MahmoodTarek — DISMISSED** [review]

File: domain/src/test/java/com/baghdad/domain/usecase/savedList/GetSavedListsUseCaseTest.kt [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    private lateinit var savedListRepository: SavedListRepository
+
+    @BeforeEach
+    fun setUp() {
+        savedListRepository = mockk(relaxed = true)
+        getSavedListsUseCase = GetSavedListsUseCase(savedListRepository)
+    }
+
+    @Test
+    fun `invoke should return paged result when repository returns data successfully`() = runTest {
```
Comment (MahmoodTarek) [inline domain/src/test/java/com/baghdad/domain/usecase/savedList/GetSavedListsUseCaseTest.kt:25]: i think you should to use that approach we Agreed about it 
`should + expected behavioral + when + condition`

**Review by MahmoodTarek — APPROVED** [review]

**Review by MuhammedEdrees — APPROVED** [review]

**Review by Arzo-zi — APPROVED** [review]
